### PR TITLE
Added Linux vnav build target

### PR DESCRIPTION
**Pull request by tridge**
This is for an arm linux board with VN-300 sensors and AHRS. This has been tested on a beaglebone black plane with VN-300

### CODE REVIEW

**Review by peterbarker — APPROVED** [review]

File: libraries/AP_HAL_Empty/RCOutput.cpp [hunk 1]
```diff
@@ -18,12 +19,23 @@ void RCOutput::disable_ch(uint8_t chan)
 {}
 
 void RCOutput::write(uint8_t chan, uint16_t period_us)
-{}
+{
+    if (chan <= ARRAY_SIZE(value)) {
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Empty/RCOutput.cpp:23]: Should be < rather than <=

File: libraries/AP_HAL_Empty/RCOutput.cpp [hunk 2]
```diff
@@ -18,12 +19,23 @@ void RCOutput::disable_ch(uint8_t chan)
...
+        value[chan] = period_us;
+    }
+}
 
-uint16_t RCOutput::read(uint8_t chan) {
+uint16_t RCOutput::read(uint8_t chan)
+{
+    if (chan <= ARRAY_SIZE(value)) {
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Empty/RCOutput.cpp:30]: Ditto

File: libraries/AP_HAL_Linux/HAL_Linux_Class.cpp [hunk 3]
```diff
@@ -163,6 +163,8 @@ static RCInput_SoloLink rcinDriver;
 static RCInput_Navio2 rcinDriver;
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
 static RCInput_RCProtocol rcinDriver{"/dev/ttyPS0", NULL};
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_VNAV
+static RCInput_RCProtocol rcinDriver{"/dev/null", NULL};
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/HAL_Linux_Class.cpp:167]: Is this a leftover or just a placeholder?

**Comment by lucasdemarchi** [comment]
Added some comments. Rest LGTM. Thanks.

File: libraries/AP_HAL_Empty/RCOutput.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_Empty/RCOutput.cpp:23]: thanks!

File: libraries/AP_HAL_Linux/HAL_Linux_Class.cpp (hunk 3 above)
Comment (tridge) [reply libraries/AP_HAL_Linux/HAL_Linux_Class.cpp:167]: it is needed to allow for rcinput via SERIALn_PROTOCOL=23 (for rcin)
ie. we need a AP_RCProtocol object, and in HAL_Linux this is how we get it

Comment (tridge) [reply libraries/AP_HAL_Linux/HAL_Linux_Class.cpp:168]: I've added a comment

Comment (lucasdemarchi) [reply libraries/AP_HAL_Linux/HAL_Linux_Class.cpp:168]: I knew this was for SERIALn_PROTOCOL. What I found strange was the use of both NULL and /dev/null (we should use either one or the other). Also neither one or the other will inject input in AP::RC().  I guess `ioctl(fd, TCSETS2, &tio)` is also failing when you try that ioctl on /dev/null and the file descriptor is closed.

Where are you injecting RC input from?  Maybe AP_RCProtocol is not the right one for this board?

Comment (lucasdemarchi) [reply libraries/AP_HAL_Linux/HAL_Linux_Class.cpp:168]: I knew this was for SERIALn_PROTOCOL. What I found strange was the use of both NULL and /dev/null (we should use either one or the other). Also neither one or the other will inject input in AP::RC().  I guess `ioctl(fd, TCSETS2, &tio)` is also failing when you try that ioctl on /dev/null and the file descriptor is closed.

Where are you injecting RC input from?  Maybe AP_RCProtocol is not the right one for this board?

**Pull request merged**